CSharp_Net_Module_1_4_Lab/CSharp_Net_Module_1_4_Lab/Program.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+
+namespace CSharp_Net_Module_1_4_Lab
+{
+    class Program
+    {
+        public const byte desktopCores = 4;
+        public const byte laptopoCores = 2;
+        public const byte serverCores = 8;
+
+        public const double desktopFrequency = 2.5;
+        public const double laptopFrequency = 1.7;
+        public const double serverFrequency = 3;
+        
+        public const int desktopMemory = 6;
+        public const int maxDesktopMemory = 8;
+        public const int laptopMemory = 4;
+        public const int serverMemory = 16;
+        
+        public const int desktopHdd = 500;
+        public const int laptopHdd  = 250;
+        public const int serverHdd = 2000;
+
+        public enum CompType { desktop, laptop, server};
+
+        public struct Cpu 
+        { 
+            public byte Cores { get;  }
+            public double Frequency { get;  }
+
+            public Cpu(byte cores, double frequency)  
+            {
+                this.Cores = cores; 
+                this.Frequency = frequency; 
+            }
+        }
+
+        public struct Computer
+        {
+            public CompType compType { get;  }
+            public Cpu CPU { get;  }
+            public int Memory { get; set;  }
+            public int HDD { get; } 
+
+            public Computer(CompType compType)
+            {
+                this.compType = compType;
+
+                switch (compType)
+                    {
+                    case CompType.desktop:
+                        this.CPU = new Cpu(desktopCores, desktopFrequency); 
+                        this.Memory = desktopMemory;
+                        this.HDD = desktopHdd; 
+                        break;
+
+                    case CompType.laptop:
+                        this.CPU = new Cpu(laptopoCores, laptopFrequency); 
+                        this.Memory = laptopMemory;
+                        this.HDD = laptopHdd;
+                        break;
+                 
+                    default: 
+                        this.CPU = new Cpu(serverCores, serverFrequency); 
+                        this.Memory = serverMemory;
+                        this.HDD = serverHdd;
+                        break;
+                }
+            }
+
+            public void UpgradeMemory(bool showComment = true)
```

Comment:
do you really need this parameter `showComment`?
as i can see you only 1 time call this method

Reply:
You are right, I dont need it. When i wrote method I thouht it needs some info about upgrading memory, but later I changed my mind and forgot  delete parameter and code that use it.